torch/csrc/jit/serialization/export_module.cpp

```diff
@@ -472,12 +474,19 @@ void ScriptModuleSerializer::writeArchive(
       &memoizedClassTypes,
       [&](const at::Tensor& tensor) {
         // returns a string to use in picker.cpp as storage obj key
-        if (tensor_cdata_naming_scheme) {
-          std::string string_id =
-              std::to_string(reinterpret_cast<std::intptr_t>(
-                  tensor.storage().unsafeGetStorageImpl()));
-          tensor_names.push_back(string_id + ".storage");
-          storage_context_.addStorage(string_id, tensor.storage());
+        if (use_storage_context) {
+          bool already_serialized =
+              storage_context_.hasStorage(tensor.storage());
+          std::string tensor_name =
+              std::to_string(
+                  storage_context_.getOrAddStorage(tensor.storage())) +
+              ".storage";
+          if (already_serialized) {
+            // this case is hit when storage has been serialized already
+            // from a torch.package context
+            serialized_tensors.insert(tensor_name);
+          }
+          tensor_names.push_back(tensor_name);
         } else {
```

Comment:
This is the part of the change that affects the mobile code

Reply:
It should also change the model generated from torch.package? It looks reasonable to me. I've run through a unit test and check one simple model. It seems ok. I may need to run some stress tests for production models. We don't have an automate way at this moment :/

Reply:
It changes the torch.package archive as well!

Reply:
Just want to double check @cccclai you ran those tests and/or are ok with the change.

Reply:
Yeah I ran the test for existing prod models and they look good from mobile side